expression-language.md

```diff
@@ -0,0 +1,369 @@
+# CloudEvents SQL Expression Language
+
+## Abstract
+
+The goal of this specification is to define an expression language SQL-like which can be used to express predicates on
+CloudEvents instances.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to Subscriptions API](#12-relation-to-the-subscriptions-api)
+
+2. [Language syntax](#2-language-syntax)
+
+- 2.1. [Expression](#21-expression)
+- 2.2. [Value identifiers and literals](#22-value-identifiers-and-literals)
+- 2.3. [Operators](#23-operators)
+- 2.4. [Functions invocation](#24-functions-invocation)
+
+3. [Language semantics](#3-language-semantics)
+
+- 3.1. [Type System](#31-type-system)
+- 3.2. [CloudEvent context identifiers and types](#32-cloudevent-context-identifiers-and-types)
+- 3.3. [Errors](#33-errors)
+- 3.4. [Operators](#34-operators)
+- 3.5. [Functions](#35-functions)
+- 3.6. [Evaluation of the expression](#36-evaluation-of-the-expression)
+
+4. [Examples](#4-examples)
+5. [References](#5-references)
+
+## 1. Introduction
+
+CloudEvents SQL expressions (also known as CESQL) allow matching of CloudEvent attributes against complex expressions
```

Comment:
Expression are going to be visible on the yaml definitions eg. trigger?

Reply:
This spec is about the language per-se, which then vendors will figure out how to integrate in their offerings